From d1d2bd4b7ccc511b5bd46169af05328154591c11 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Tue, 08 Jun 2010 21:56:22 +0000
Subject: Fix build for --disable-smartcard-support

commit 4b334ddf6c53ce795063d44746a9a2d90fea85ff broke the
build for --disable-smartcard-users.

This commit fixes that.  Based on a patch from
William Jon McCann <william.jon.mccann@gmail.com>

https://bugzilla.gnome.org/show_bug.cgi?id=617748
---
diff --git a/plugins/Makefile.am b/plugins/Makefile.am
index 027ae16..ef82aa0 100644
--- a/plugins/Makefile.am
+++ b/plugins/Makefile.am
@@ -1,7 +1,6 @@
 NULL =
 
-SUBDIRS =		\
-	common		\
+enabled_plugins =	\
 	a11y-keyboard	\
 	background	\
 	clipboard	\
@@ -12,10 +11,20 @@ SUBDIRS =		\
 	keyboard	\
 	media-keys	\
 	mouse		\
-	smartcard	\
 	sound		\
 	typing-break	\
 	xrandr		\
 	xrdb		\
 	xsettings	\
 	$(NULL)
+
+disabled_plugins = $(NULL)
+
+if SMARTCARD_SUPPORT
+enabled_plugins += smartcard
+else
+disabled_plugins += smartcard
+endif
+
+SUBDIRS = common $(enabled_plugins)
+DIST_SUBDIRS = $(SUBDIRS) $(disabled_plugins)
diff --git a/plugins/smartcard/Makefile.am b/plugins/smartcard/Makefile.am
index 62e253e..dcfaf49 100644
--- a/plugins/smartcard/Makefile.am
+++ b/plugins/smartcard/Makefile.am
@@ -1,4 +1,3 @@
-if SMARTCARD_SUPPORT
 plugin_LTLIBRARIES = \
 	libsmartcard.la
 
@@ -32,7 +31,6 @@ libsmartcard_la_LIBADD = \
 	$(NSS_LIBS)
 
 @GSD_INTLTOOL_PLUGIN_RULE@
-endif
 
 plugin_in_files = \
 	smartcard.gnome-settings-plugin.in
--
cgit v0.8.3.1
